fpga.chart.py

```diff
@@ -5,152 +5,161 @@
-                  ['current_pu_utilised_decomp_percent', 'current decompress PUs (%)', 'absolute'],
-                  ['avg_pu_utilised_comp_percent', 'avg compress PUs (%)', 'absolute'],
-                  ['avg_pu_utilised_decomp_percent', 'avg decompress PUs (%)', 'absolute'],
-                  ['max_pu_utilised_comp', 'max. compress PUs', 'absolute'],
-                  ['max_pu_utilised_decomp', 'max. decompress PUs', 'absolute']
-                 ]
+        'lines': [
+            ['current_pu_utilised_comp_percent', 'current compress PUs (%)', 'absolute'],
+            ['current_pu_utilised_decomp_percent', 'current decompress PUs (%)', 'absolute'],
+            ['avg_pu_utilised_comp_percent', 'avg compress PUs (%)', 'absolute'],
+            ['avg_pu_utilised_decomp_percent', 'avg decompress PUs (%)', 'absolute'],
+            ['max_pu_utilised_comp', 'max. compress PUs', 'absolute'],
+            ['max_pu_utilised_decomp', 'max. decompress PUs', 'absolute']
+        ]
     },
     'ddr_stats': {
         'options': [None, 'Successful and denied DDR transfers', 'Transfers', 'fpga', 'fpga', 'line'],
-        'lines': [['avg_memory_write_transactions_percent', 'successful write transfers (%)', 'absolute'],
-                  ['avg_memory_read_transactions_percent', 'successful read transfers (%)', 'absolute'],
-                  ['avg_memory_write_denied_percent', 'denied write transfers (%)', 'absolute'],
-                  ['avg_memory_read_denied_percent', 'denied read transfers (%)', 'absolute']
-                 ]
+        'lines': [
+            ['avg_memory_write_transactions_percent', 'successful write transfers (%)', 'absolute'],
+            ['avg_memory_read_transactions_percent', 'successful read transfers (%)', 'absolute'],
+            ['avg_memory_write_denied_percent', 'denied write transfers (%)', 'absolute'],
+            ['avg_memory_read_denied_percent', 'denied read transfers (%)', 'absolute']
+        ]
     }
 }
 
+def make_name(prefix, postfix):
+    return prefix + '-' + postfix
+
 class Service(SimpleService):
     def __init__(self, configuration=None, name=None):
         SimpleService.__init__(self, configuration=configuration, name=name)
+
         self.conn = None
         self.default_data = dict()
         self.order = []
         self.definitions = dict()
-        self.keys = []
-        self.fpga_mapping = dict()
-        self.next_fpga = 0
-        self.fpga_count = 1
         self.dsn = self.configuration.get('dsn')
-        self.pu_ddr_stats_enable = self.configuration.get('pu_ddr_stats_enable')
-        self.metrics = [ 'bytes', 'jobs', 'max' ]
 
-        conn = self._connect(self.dsn)
-        with conn.cursor() as cursor:
-            cursor.execute('CREATE EXTENSION IF NOT EXISTS swarm64da')
-            cursor.execute('SELECT COUNT(*) FROM swarm64da.get_fpga_stats()')
-            self.fpga_count = cursor.fetchone()[0]
+        self.fpga_mapping = dict()
```

Comment:
Minor: please use `{}` instead of `dict()`. See eg. https://doughellmann.com/blog/2012/11/12/the-performance-impact-of-using-dict-instead-of-in-cpython-2-7-2/

Reply:
Makes sense, thought that was a Python 2.7 incompatibility or so.